src/main/java/net/dries007/tfc/client/ClientEventHandler.java

```diff
@@ -106,7 +112,10 @@ public static void registerColorHandlerBlocks(ColorHandlerEvent.Block event)
         registry.register((state, worldIn, pos, tintIndex) -> TFCColors.getGrassColor(pos, tintIndex), TFCBlocks.SOIL.get(SoilBlockType.CLAY_GRASS).values().stream().map(RegistryObject::get).toArray(Block[]::new));
         registry.register((state, worldIn, pos, tintIndex) -> TFCColors.getGrassColor(pos, tintIndex), TFCBlocks.PEAT_GRASS.get());
         // Plants
-        registry.register((state, worldIn, pos, tintIndex) -> TFCColors.getGrassColor(pos, tintIndex), TFCBlocks.PLANTS.values().stream().map(RegistryObject::get).toArray(Block[]::new));
+        Block[] leafyPlants = Stream.of(Plant.values()).filter(Plant::isLeafColored).map(p -> TFCBlocks.PLANTS.get(p).get()).toArray(Block[]::new);
+        Block[] grassyPlants = Stream.of(Plant.values()).filter(p -> !p.isLeafColored()).map(p -> TFCBlocks.PLANTS.get(p).get()).toArray(Block[]::new);
+        registry.register((state, worldIn, pos, tintIndex) -> TFCColors.getSeasonalFoliageColor(state, pos, tintIndex, Wood.Default.HICKORY.getFallFoliageCoords()), leafyPlants);
```

Comment:
Having this delegate to hickory specifically is a bit weird imo, either move that to a static reference for plants or add `Plant#getFallFoliageCoords()`